core-web/libs/template-builder/src/lib/components/template-builder/components/add-style-classes-dialog/add-style-classes-dialog.component.spec.ts

```diff
@@ -0,0 +1,142 @@
+    let component: AddStyleClassesDialogComponent;
+    let spectator: SpectatorHost<AddStyleClassesDialogComponent>;
+
+    const createHost = createHostFactory({
+        component: AddStyleClassesDialogComponent,
+        imports: [
+            AutoCompleteModule,
+            FormsModule,
+            NoopAnimationsModule,
+            ButtonModule,
+            HttpClientModule
+        ],
+        providers: [
+            {
+                provide: DynamicDialogConfig,
+                useValue: {
+                    data: {
+                        selectedClasses: MOCK_SELECTED_STYLE_CLASSES
+                    }
+                }
+            },
+            {
+                provide: DotAddStyleClassesDialogStore,
+                useValue: addStyleClassesStoreMock
+            },
+            DynamicDialogRef
+        ]
+    });
+
+    beforeEach(() => {
+        spectator = createHost(
+            '<dotcms-add-style-classes-dialog></dotcms-add-style-classes-dialog>',
+            {}
+        );
+
+        component = spectator.component;
+        spectator.detectChanges();
+
+        Object.defineProperty(window, 'matchMedia', {
+            writable: true,
+            value: jest.fn().mockImplementation((query) => ({
+                matches: false,
+                media: query,
+                onchange: null,
+                addListener: jest.fn(), // Deprecated
+                removeListener: jest.fn(), // Deprecated
+                addEventListener: jest.fn(),
+                removeEventListener: jest.fn(),
+                dispatchEvent: jest.fn()
+            }))
+        });
+    });
+
+    it('should have an autocomplete component', () => {
+        expect(spectator.query('p-autocomplete')).toBeTruthy();
+    });
+
+    it('should have an update button', () => {
+        expect(spectator.query(byTestId('update-btn'))).toBeTruthy();
+    });
+
+    it('should trigger filterMock when focusing on the input', () => {
+        const filterMock = jest.spyOn(component, 'filterClasses');
+
+        const input = spectator.query('#my-autocomplete') as HTMLInputElement;
+
+        spectator.click(input);
+
+        spectator.detectChanges();
+
+        expect(filterMock).toHaveBeenCalled();
+    });
+
+    it('should trigger addClassByCommaOrSpace when typing on the input and adding comma', () => {
+        const addMock = jest.spyOn(component, 'addClassByCommaOrSpace');
+
+        const input = spectator.query('#my-autocomplete') as HTMLInputElement;
+
+        input.value = 'd-none,';
+
+        spectator.click(input);
+
+        spectator.detectChanges();
+
+        expect(addMock).toHaveBeenCalledWith('d-none,');
+        expect(component.selectedClasses).toContainEqual({ cssClass: 'd-none' });
+    });
+
+    it('should trigger addClassByCommaOrSpace when typing on the input and adding space', () => {
+        const addMock = jest.spyOn(component, 'addClassByCommaOrSpace');
```

Comment:
You should not spy the methods of the component you are testing, you are only testing that this is being called but you should test what actually do the method.